src/playwrightTestServer.ts

```diff
@@ -427,10 +427,10 @@ export class PlaywrightTestServer {
     return new Promise<void>(resolve => {
       const disposables = [
         testServer.onReport(async message => {
-          if (token.isCancellationRequested && message.method !== 'onEnd')
+          if (token.isCancellationRequested && message.method !== 'onExit')
             return;
           await teleReceiver.dispatch(message);
-          if (message.method === 'onEnd') {
+          if (message.method === 'onExit') {
```

Comment:
Perhaps, instead of this, we should actually wait for the `TestServer.runTests()` to finish?

Reply:
Good idea! I also took the liberty to refactor some of the resource cleanup and make it easier to follow.